src/hotspot/share/opto/type.cpp

```diff
@@ -5006,12 +5006,12 @@ void TypeAryPtr::dump2( Dict &d, uint depth, outputStream *st ) const {
   }
 
   if( _offset != 0 ) {
-    int header_size = objArrayOopDesc::header_size() * wordSize;
+    BasicType basic_elem_type = elem()->basic_type();
+    int header_size = arrayOopDesc::base_offset_in_bytes(basic_elem_type);
```

Comment:
This function now calls base_offset_in_bytes twice. One to set header_size and the other to set the array_base. Maybe clean that up?